book/src/fork-selection.md

```diff
@@ -0,0 +1,147 @@
+# Fork Selection
+This chapter defines Solana's *Nakomoto Fork Selection* algorithm based on time
+locks. It satisfies the following properties:
+
+
+* A voter can eventually recover from voting on a fork that doesn't become the
+  fork with the desired finality.
+* If the voters share a common ancestor then they will converge to a fork
+  containing that ancestor no matter how they are partitioned, although it may
+  not be the latest possible ancestor at the start of the fork.
+* Rollback requires exponentially more time for older votes than for newer
+  votes.
+* Voters have the freedom to set a minimum finality threshold before committing
+  a vote to a higher lockout.  This allows each voter to make a trade-off
+  between risk and reward. See [finality](#economic-finality).
+
+## Time
+
+For networks like Solana, time can be the PoH hash count, which is a VDF that
+provides a source of time before consensus. Other networks adopting this
+approach would need to consider a global source of time.
+
+For Solana, time uniquely identifies a specific leader for fork generation.  At
+any given time only 1 leader, which can be computed from the ledger itself, can
+propose a fork.  For more details, see [fork generation](fork-generation.md)
+and [leader rotation](leader-rotation.md).
+
+## Algorithm
+
+The basic idea to this approach is to stack consensus votes.  Votes in the
+stack must be for forks that descend from each other and for forks that are
```

Comment:
"descend from each other" -> "share a common ancestor"?

Reply:
@garious each vote on the stack is an ancestor of the vote above it.